packages/@lwc/style-compiler/src/serialize.ts

```diff
@@ -107,7 +107,11 @@ function reduceTokens(tokens: Token[]): Token[] {
         .reduce((acc: Token[], token: Token) => {
             const prev = acc[acc.length - 1];
             if (token.type === TokenType.text && prev && prev.type === TokenType.text) {
-                prev.value += token.value;
+                // clone the previous token to avoid mutating it in-place
+                acc[acc.length - 1] = {
+                    type: prev.type,
+                    value: prev.value + token.value,
+                };
```

Comment:
We might not technically need this, but I found it annoying during local testing that `generateExpressionFromTokens` actually mutates the input.